### PR TITLE
Add GLB drag support

### PR DESCRIPTION
**Pull request by uhunkler**
Hi Vincent,

I am missing the Hubs Blender Plugin easy GLB workflow. So I took your ticket #32 »Implement drag and drop scene glb for development #32« and added Drag and Drop in index.html. Please have a look, if the implementation is ok for you.

### CODE REVIEW

File: examples/playground/index.html [hunk 1]
```diff
@@ -173,5 +173,37 @@
...
+          e.dataTransfer.dropEffect = "copy";
+      }, false);
+
+      document.addEventListener("drop", function(e) {
+          e.stopPropagation();
+          e.preventDefault();
+
+          const files = e.dataTransfer.files; // Array of all files.
+
+          for (let i = 0, file; file = files[i]; i++) {
```
Comment (vincentfretin) [inline examples/playground/index.html:191]: I prefer the more modern for-of syntax. Can you please try
` for (const file of files) {`

File: examples/playground/index.html [hunk 2]
```diff
@@ -173,5 +173,37 @@
...
+              // Process only GLB files.
+              if (file.name.match(/\.(glb)$/)) {
```
Comment (vincentfretin) [inline examples/playground/index.html:193]: I avoid regex when I can. Can you please use `file.name.endsWith(".glb")`

**Review by vincentfretin — CHANGES_REQUESTED** [review]
Thank you! I added two comments. If you can change that, it would be good to merge.

**Comment by uhunkler** [comment]
I changed the two lines as you proposed. Is the code ok now?

File: examples/playground/index.html [hunk 3]
```diff
@@ -173,5 +173,37 @@
...
+              // Process only GLB files.
+              if (file.name.endsWith(".glb")) {
+                  const reader = new FileReader();
+
+                  reader.onload = (function(theFile) {
+                      return function(e) {
+                          // Set the gltf-model-plus attribute of the gltf-model-plus component to the dropped file.
+                          document.querySelector("[gltf-model-plus]").setAttribute("gltf-model-plus", e.target.result);
+                      };
+                  })(file);
```
Comment (vincentfretin) [inline examples/playground/index.html:201]: `function(theFile)` doesn't do anything here, you don't use theFile.
You just need:
```js
             reader.onload = function (e) {
                // Set the gltf-model-plus attribute of the gltf-model-plus component to the dropped file.
                document.querySelector("[gltf-model-plus]").setAttribute("gltf-model-plus", e.target.result);
              };
```

Can you please change it and reformat the file with prettier?

**Review by vincentfretin — CHANGES_REQUESTED** [review]

File: examples/playground/index.html [hunk 4]
```diff
@@ -201,5 +201,43 @@ <h5 style="margin-top: 0.1rem; margin-bottom: 0.2rem">Help</h5>
...
+        "drop",
+        function (e) {
+          e.stopPropagation();
+          e.preventDefault();
+
+          const files = e.dataTransfer.files; // Array of all files.
+
+          for (const file of files) {
+            // Process only GLB files.
+            if (file.name.endsWith(".glb")) {
```
Comment (jywarren) [inline examples/playground/index.html:227]: Maybe we should recognize .bin files or leave a notification to people that their Hubs scene files may be named .bin but be glb format... I could submit a PR if helpful.

Comment (vincentfretin) [reply examples/playground/index.html:227]: It's not really standard to name a glb .bin. What is standard is a .gltf with a .bin, not .bin alone. I hope the official hubs export will name the file .glb. You also need to rename to .glb if you want to test it on https://gltf-viewer.donmccurdy.com/

**Pull request merged**